entity-framework/core/miscellaneous/async.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Asynchronous Programming - EF Core
+description: Querying and saving data asynchronously with Entity Framework Core
+author: roji
+ms.date: 9/2/2020
+ms.assetid: 38f71624-7a68-4c72-a918-3e7b858ef090
+uid: core/miscellaneous/async
+---
+# Asynchronous Programming
+
+Asynchronous queries avoid blocking a thread while the query is executed in the database. Async queries are important for keeping a responsive UI in thick-client applications, and can also increase throughput in web applications where they free up the thread to service other requests in web applications.
+
+Following the .NET standard, EF Core provides asynchronous counterparts to all synchronous methods which perform I/O. These have the same effects as the sync methods, and can be used with the C# `async` and `await` keywords. For example, instead of using DbContext.SaveChanges, which will block a thread while database I/O is performed, DbContext.SaveChangesAsync can be used:
+
+[!code-csharp[Main](../../../samples/core/Miscellaneous/Async/Program.cs#SaveChangesAsync)]
+
+For more information,, see [the general C# asynchronous programming docs](/dotnet/csharp/async).
+
+> [!WARNING]
+> EF Core doesn't support multiple parallel operations being run on the same context instance. You should always wait for an operation to complete before beginning the next operation. This is typically done by using the `await` keyword on each async operation.
+
+# Async LINQ operators
+
+In order to support executing LINQ queries asynchronously, EF Core provides a set of async extension methods which execute the query and return results. These counterparts to the standard, synchronous LINQ operators include ToListAsync, SingleAsync, AsAsyncEnumerable, etc.:
+
+[!code-csharp[Main](../../../samples/core/Miscellaneous/Async/Program.cs#ToListAsync)]
+
+Note that there are no async versions of some LINQ operators such as Where or OrderBy, because these only build up the LINQ expression tree and don't cause the query to be executed in the database. Only operators which cause query execution have async counterparts.
+
+> [!IMPORTANT]
+> The EF Core async extension methods are defined in the `Microsoft.EntityFrameworkCore` namespace. This namespace must be imported for the methods to be available.
+
+# Client-side async LINQ operators
+
+The async LINQ operators discussed above can only be used on EF queries - you cannot use them with client-side LINQ to Objects query. To perform client-side async LINQ operations outside of EF, use the [System.Interactive.Async package](https://www.nuget.org/packages/System.Interactive.Async); this can be especially useful for performing operations on the client that cannot be translated for evaluation at the server.
```

Comment:
avoid this without noun

Reply:
I think it's fine, but added "package".

Reply:
Guidance of doc tool is to avoid this without noun since it can be ambiguous what it refers to. While for us since we know product and also "native" English speaker, it is easy to understand, it can be difficult for others.